xla/service/gpu/ir_emission_utils.cc

```diff
@@ -740,6 +739,90 @@ bool IsIntermediate(const HloInstruction* instr, int allowed_operand_count) {
                                       instr->shape(), instr->dimensions()))));
 }
 
+bool IsReduceIntermediate(const HloInstruction* instr) {
```

Comment:
Can this function be moved to reduction_utils or gpu_fusible?

Reply:
yea i think reduction_utils might be a better place for this